src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
The problem is that the db case is an example and I can see other cases where it would be needed.
I'm pretty sure there will be many cases where it will happen…
The alternative, is to stay in the dw's style and use a Factory and all BUT we would still use the ConfigSingleton provided by pac4j in order to give access to the Config for those who wants…
It would avoid making people do akward things where they manipulate dw configuration in the run method…

I will take a look at dropwizard-db in details, I don't use it personally, I build a mongodb client (I made my own MongoConfigurationFactory with a build method and all) in the run method, so it is not available before! But I haven't looked yet at dropwizard-db, so maybe there is something more elegant yes :)